scripts/rpc/getSaleHistory.ts

```diff
@@ -0,0 +1,70 @@
+/*
+ * An example of how to query the Foundation sale history of an NFT using RPC (e.g. Alchemy) requests.
+ * Run with `yarn run-script scripts/rpc/getSaleHistory.ts --network mainnet`
+ */
+
+import { ethers } from "hardhat";
+import { FNDNFTMarket__factory } from "../../typechain-types";
+import { ReserveAuctionFinalizedEvent } from "../../typechain-types/FNDNFTMarket";
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const addresses = require("@f8n/fnd-protocol/addresses.js");
+
+// The example NFT to query for, this should be updated with the NFT of interest
+const collection = "0x3B3ee1931Dc30C1957379FAc9aba94D1C48a5405";
+const tokenId = 105912;
+
+async function main(): Promise<void> {
+  // Connect to the Foundation market contract
+  const market = FNDNFTMarket__factory.connect(addresses.prod[1].nftMarket, ethers.provider);
+
+  // Query for all sales via Buy Price and via Offers
+  const buyPricesAccepted = await market.queryFilter(market.filters.BuyPriceAccepted(collection, tokenId));
+  const offersAccepted = await market.queryFilter(market.filters.OfferAccepted(collection, tokenId));
+  // It is possible to query for all buyPricesAccepted and offersAccepted in a single request, but for simplicity we do it separately here
+
+  const auctionsCreated = await market.queryFilter(market.filters.ReserveAuctionCreated(null, collection, tokenId));
```

Comment:
These seem pretty generic and useful all around so not sure if its an overkill but might be nice to refactor these into a helper fn like getFinalizedEventsByContractTokenId(marketToolType, contractAddress, TokenId) for future use.

Reply:
Yea maybe. Something like that could be useful in the NPM package too.

But for an example we might reference, not using helpers much actually makes it easier to read & copy paste from.